### ISSUE TITLE
`deno compile` failed to exec cjs format entrance

### ISSUE THREAD

**Issue by greenhat616** [denoland/deno#26388]
Version: Deno 2.0.2

If a run the cjs file directly:
```shell
deno run -A ./sub-store.bundle.cjs
This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.

┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
     Sub-Store -- v2.14.401
┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅

[sub-store] INFO: [BACKEND] :::3000
```

It works as expected.

However, when i compile it into a single file:
```
deno compile -A ./sub-store.bundle.cjs
```

It refuses to exec:
```
# ./sub-store.bundle -v
error: Uncaught Error: Cannot find module '/tmp/deno-compile-sub-store.bundle/backend/dist/sub-store.bundle.cjs'
    at Function.Module._resolveFilename (node:module:622:15)
    at Function.Module._load (node:module:500:27)
    at Object.loadCjsModule (ext:deno_node/02_init.js:64:22)
    at loadCjsModule ([ext:ext/node/lib.rs:782:29]:2:32)
    at [ext:ext/node/lib.rs:782:29]:3:7
```

**Comment by yazan-abdalrahman** [comment]
@greenhat616 
It is not reproducible on Windows. Perhaps you could provide me the reproduce steps since I may not have followed them correctly. 


```
PS D:\yad\deno> cat .\sub-store.bundle.cjs                            
console.log("hello_test");
PS D:\yad\deno> 
PS D:\yad\deno> .\target\debug\deno  run -A ./sub-store.bundle.cjs    
hello_test
PS D:\yad\deno> 
PS D:\yad\deno> .\target\debug\deno compile  -A ./sub-store.bundle.cjs
Compile file:///D:/yad/deno/sub-store.bundle.cjs to sub-store.bundle.exe
PS D:\yad\deno> 
PS D:\yad\deno>  .\sub-store.bundle.exe                               
hello_test
PS D:\yad\deno> 
PS D:\yad\deno>  .\sub-store.bundle.exe -v
hello_test
PS D:\yad\deno> 

```

**Comment by greenhat616** [comment]
@yazan-abdalrahman I can confirm it was fixed on 2.0.3
![image](https://github.com/user-attachments/assets/cb694961-924e-4565-881d-393e901d8569)
